qiskit/algorithms/gradients/finite_diff_estimator_gradient.py

```diff
@@ -0,0 +1,99 @@
+    """
+    Compute the gradients of the expectation values by finite difference method.
+    """
+
+    def __init__(self, estimator: BaseEstimator, epsilon: float, **run_options):
+        """
+        Args:
+            estimator: The estimator used to compute the gradients.
+            epsilon: The offset size for the finite difference gradients.
+            run_options: Backend runtime options used for circuit execution. The order of priority is:
+                run_options in ``run`` method > gradient's default run_options > primitive's default
+                setting. Higher priority setting overrides lower priority setting.
+
+        Raises:
+            ValueError: If ``epsilon`` is not positive.
+        """
+        if epsilon <= 0:
+            raise ValueError(f"epsilon ({epsilon}) should be positive.")
+        self._epsilon = epsilon
+        self._base_parameter_values_dict = {}
+        super().__init__(estimator, **run_options)
+
+    def _run(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        observables: Sequence[BaseOperator | PauliSumOp],
+        parameter_values: Sequence[Sequence[float]],
+        parameters: Sequence[Sequence[Parameter] | None],
+        **run_options,
+    ) -> EstimatorGradientResult:
+        """Compute the estimator gradients on the given circuits."""
+        jobs, metadata_ = [], []
+        for circuit, observable, parameter_values_, parameters_ in zip(
+            circuits, observables, parameter_values, parameters
+        ):
+            # indices of parameters to be differentiated
+            if parameters_ is None:
+                indices = list(range(circuit.num_parameters))
+            else:
+                indices = [circuit.parameters.data.index(p) for p in parameters_]
+            metadata_.append({"parameters": [circuit.parameters[idx] for idx in indices]})
+
+            offset = np.identity(circuit.num_parameters)[indices, :]
+            plus = parameter_values_ + self._epsilon * offset
+            minus = parameter_values_ - self._epsilon * offset
+            n = 2 * len(indices)
+
+            job = self._estimator.run(
+                [circuit] * n, [observable] * n, plus.tolist() + minus.tolist(), **run_options
+            )
+            jobs.append(job)
+
+        # combine the results
+        if any(job.status() is not JobStatus.DONE for job in jobs):
+            raise QiskitError("The gradient job was not completed successfully. ")
+
+        results = [job.result() for job in jobs]
+        gradients = []
+        for result in results:
+            n = len(result.values) // 2  # is always a multiple of 2
+            gradient_ = (result.values[:n] - result.values[n:]) / (2 * self._epsilon)
+            gradients.append(gradient_)
+
+        # TODO: include primitive's run_options as well
+        return EstimatorGradientResult(
+            gradients=gradients, metadata=metadata_, run_options=run_options
```

Comment:
In Fidelties the metadata here is what the underlying primitive gives back. Since we have several jobs for a single gradient perhaps any metadata from there is not relevant to be passed back.